rfd/0179-api-versioning.md

```diff
@@ -146,6 +146,37 @@ func (h *Handler) getFoos() (interface{}, error) {
 }
 ```
 
+#### Defining new endpoints in the web client
+All endpoints should be defined in a JS object, similary done [here](https://github.com/gravitational/teleport.e/blob/ebf079267020d59f14353dd3495b4fd783339fa5/web/teleport/src/config.ts#L134).
+
+This helps us tell apart same paths but with different HTTP verbs.
+
+Example of a new single endpoint:
+
+```ts
+user: {
+  create: '/v1/webapi/users',
+}
+```
+
+Example of an endpoint with same paths but with different verbs:
+
+```ts
+user: {
+  create: '/v1/webapi/users',
+  update: '/v1/webapi/users',
+}
+```
+
+Example of creating a version N endpoint:
+```ts
+user: {
+  // TODO(<your-github-handle>): DELETE IN 18.0 - replaced by /v2/webapi/users
+  create: '/v1/webapi/users',
+  createV2: '/v2/webapi/users',
+}
```

Comment:
I think we'd only need the `verbV2` if we plan on using both (like a fallback for example). Otherwise, the current client should just have the version it plans on using. 

So I think this specific pattern is fine, I just don't think we'll be using it too often

Reply:
got it, i think you wanted me to leave some comment about marking it for deletion only if we need a fallback right? which i did here: https://github.com/gravitational/teleport/pull/50743/files#diff-7852f06643f466fb42c702c6f8bb51f67339dbb53e40629bce1c2a69918d1d0dR188